### PR TITLE
feat: Use RoutesBuilder API

### PR DESCRIPTION
**Pull request by mshabarov**

### CODE REVIEW

File: src/main/frontend/routes.tsx [hunk 1]
```diff
@@ -1,24 +1,13 @@
...
-            { path: '/hilla', element: <Hilla />, handle: { title: 'Hilla', rolesAllowed: ['ROLE_USER'] } }
-        ],
-    },
-    { path: '/login', element: <Login />, handle: { title: 'Login' } }
-]) as RouteObject[];
-export const routes = buildRoute(routing, routing[0].children);
-
-export default createBrowserRouter(routes);
\ No newline at end of file
+export const { routes, router} = new RouterConfigurationBuilder()
```
Comment (tltv) [inline src/main/frontend/routes.tsx]:
```suggestion
export const { routes, router } = new RouterConfigurationBuilder()
```

File: src/main/frontend/routes.tsx [hunk 2]
```diff
@@ -1,24 +1,13 @@
...
-import { buildRoute } from "Frontend/generated/flow/Flow";
-import Layout from "Frontend/views/$layout";
-import Hilla from "Frontend/views/hilla";
-import About from "Frontend/views/about";
+import Flow from 'Frontend/generated/flow/Flow';
+import fileRoutes from 'Frontend/generated/file-routes';
 import Login from "Frontend/views/login";
-import {protectRoutes} from "@vaadin/hilla-react-auth";
-import Public from "Frontend/views/$index";
+import {RouterConfigurationBuilder} from "@vaadin/hilla-file-router/runtime.js";
```
Comment (tltv) [inline src/main/frontend/routes.tsx]:
```suggestion
import { RouterConfigurationBuilder } from "@vaadin/hilla-file-router/runtime.js";
```

File: src/main/frontend/views/about.tsx [hunk 3]
```diff
@@ -1,4 +1,9 @@
 import { VerticalLayout } from "@vaadin/react-components/VerticalLayout.js";
+import type {ViewConfig} from "@vaadin/hilla-file-router/types.js";
```
Comment (tltv) [inline src/main/frontend/views/about.tsx]:
```suggestion
import type { ViewConfig } from "@vaadin/hilla-file-router/types.js";
```

File: src/main/frontend/views/hilla.tsx [hunk 4]
```diff
@@ -3,6 +3,12 @@ import { TextField } from "@vaadin/react-components/TextField.js";
 import { HelloEndpoint } from "Frontend/generated/endpoints";
 import { useState } from "react";
 import {VerticalLayout} from "@vaadin/react-components/VerticalLayout.js";
+import type {ViewConfig} from "@vaadin/hilla-file-router/types.js";
```
Comment (tltv) [inline src/main/frontend/views/hilla.tsx]:
```suggestion
import type { ViewConfig } from "@vaadin/hilla-file-router/types.js";
```

File: src/main/frontend/views/hilla.tsx [hunk 5]
```diff
@@ -3,6 +3,12 @@ import { TextField } from "@vaadin/react-components/TextField.js";
 import { HelloEndpoint } from "Frontend/generated/endpoints";
 import { useState } from "react";
 import {VerticalLayout} from "@vaadin/react-components/VerticalLayout.js";
```
Comment (tltv) [inline src/main/frontend/views/hilla.tsx]:
```suggestion
import { VerticalLayout } from "@vaadin/react-components/VerticalLayout.js";
```

**Review by tltv — APPROVED** [review]
Suggesting import format to be same everywhere with a spaces around inside `import { ... }`.

**Review by caalador — COMMENTED** [review]
Looks fine, but it works a bit strange.
It had all the routes for the different users, also the ones that are not applicable for the user. 
So the Flow view is there for user, which leads to a `Reason: Access is denied by annotations on the view.`.
And Hilla User is there for admin which leads to the login view, even though you are already logged in.

**Comment by mshabarov** [comment]
> It had all the routes for the different users, also the ones that are not applicable for the user.
So the Flow view is there for user, which leads to a Reason: Access is denied by annotations on the view..
And Hilla User is there for admin which leads to the login view, even though you are already logged in.

Yeah, I'll try to change that by editing the main layout impl.

**Comment by mshabarov** [comment]
Unfortunately, I couldn't manage to make it work with the `createMenuItems` method, because:

1. Any item returned by `createMenuItems` seems to not have `rolesAllowed`, nor `loginRequired` properties, I don't know how could I get the authorities with this API
2. Authorities for server-side views aren't sent to the client and basically they aren't filtered by server (this feature haven't been implemented yet).

Thus, I'm not sure how can I filter the menu items by access (in role and is logged in).

I pushed an ugly checking codes that works, but we have to replace it with a proper way.

**Comment by mshabarov** [comment]
UPD:
`createMenuItems` function now checks the access control automatically, but this feature hasn't been released in any alpha yet. Should be available in Hilla 24.4.0.alpha22, and then I can change my main layout impl to use:

```typescript
<nav>
    {
         createMenuItems().map(({ to, icon, title }) => (
                <NavLink className={navLinkClasses} to={to} key={to}>
                      {title}
                </NavLink>
              ))
         },
         { flowIsInRole ? (
                <NavLink to={'/flow'} className={navLinkClasses}>
                      Flow Admin
                </NavLink>
         ) : null}
</nav>
```

Later on, when we implement [this feature](https://github.com/vaadin/flow/issues/19127) in Flow, we will be able to even remove that part 
```typescript
         { flowIsInRole ? (
                <NavLink to={'/flow'} className={navLinkClasses}>
                      Flow Admin
                </NavLink>
         ) : null}
```
as server-side routes will be checked agains access control and added into `createMenuItems` output list, if have access.

So we should wait for Vaadin 24.4.0.alpha22 release, remove my workaround for Hilla view, then wait for the above feature in Flow and just leave the following codes:
```typescript
<nav>
    {
         createMenuItems().map(({ to, icon, title }) => (
                <NavLink className={navLinkClasses} to={to} key={to}>
                      {title}
                </NavLink>
              ))
         }
</nav>
```
FYI @caalador @tltv

**Pull request merged**